packages/nextjs/app/builders/page.tsx

```diff
@@ -0,0 +1,217 @@
+              <div
+                key={id}
+                className="card bg-base-100 shadow hover:shadow-md transition-all duration-300 overflow-hidden border border-transparent hover:border-primary"
+              >
+                <div className="card-body p-3">
+                  <div className="flex items-center gap-3 mb-1">
+                    {/* Avatar image */}
+                    <div className="avatar">
+                      <div className="w-10 h-10 rounded-full ring ring-primary ring-offset-base-100 ring-offset-1 overflow-hidden">
+                        {builder.avatarUrl &&
+                          (builder.avatarUrl.startsWith("/") ? (
+                            <Image
+                              src={builder.avatarUrl}
+                              alt={`${builder.name}'s avatar`}
+                              width={48}
+                              height={48}
+                              className="w-full h-full object-cover"
+                            />
+                          ) : (
+                            <Image
+                              src={builder.avatarUrl}
+                              alt={`${builder.name}'s avatar`}
+                              width={48}
+                              height={48}
+                              className="w-full h-full object-cover"
+                            />
+                          ))}
+                      </div>
+                    </div>
+
+                    <div>
+                      <h3 className="font-bold text-sm">{builder.name}</h3>
+                      <div className="text-xs truncate">
+                        <Address address={builder.address} />
+                      </div>
+                    </div>
+                  </div>
+
+                  <div className="card-actions justify-end mt-1">
+                    {builder.hasPersonalPage ? (
+                      <Link href={builder.profilePath || `#`} className="btn btn-xs btn-primary">
+                        View
+                      </Link>
+                    ) : (
+                      <button className="btn btn-xs btn-outline" disabled>
+                        No Profile
+                      </button>
+                    )}
+                  </div>
+                </div>
+              </div>
+            ))}
+          </div>
+        )}
+
+        {/* Empty State */}
+        {!isLoading && builders.length === 0 && (
+          <div className="text-center py-16">
+            <h1 className="text-5xl font-extrabold mb-4 animate-text bg-gradient-to-r from-purple-600 via-pink-500 to-blue-500 bg-clip-text text-transparent">
+              Meet the Builders of Batch 14
+            </h1>
+
+            <p className="mt-6 text-xl text-gray-600 max-w-2xl mx-auto">
+              The brightest minds building the future of web3. Explore their profiles and see what they&apos;re
+              creating!
+            </p>
+          </div>
+        )}
```

Comment:
This text flashes for half a second before the builders are shown. Very confusing.